website/docs/r/service.html.markdown

```diff
@@ -3,63 +3,76 @@ layout: "consul"
-A high-level resource for creating a Service in Consul. Currently,
-defining health checks for a service is not supported.
+A high-level resource for creating a Service in Consul in the Consul catalog. This
+is appropriate for registering [external services](https://www.consul.io/docs/guides/external.html) and
+can be used to create services addressable by Consul that cannot be registered
+with a [local agent](https://www.consul.io/docs/agent/basics.html).
 
-**Most users should not use this resource**. When using Consul with
-compute instances, it's better to install
-[the Consul Agent](https://www.consul.io/docs/agent/basics.html)
-on these machines and register services via the agent. This ensures
-that services get assigned to the appropriate Consul "nodes" and
-allows service health to integrate with general node health as
-reported by the agent.
-
-To register a non-compute resource, such as a hosted database,
-as a service, as described in
-[Consul's _External Services_ guide](https://www.consul.io/docs/guides/external.html),
-use [`consul_catalog_entry`](catalog_entry.html) instead, which
-can create an arbitrary service record in the Consul catalog.
+If the Consul agent is running on the node where this service is registered, it is
+not recommended to use this resource.
 
 ## Example Usage
 
+Creating a new node with the service:
+
 ```hcl
 resource "consul_service" "google" {
-  address = "www.google.com"
   name    = "google"
+  node    = "${consul_node.compute.name}"
   port    = 80
-  tags    = ["tag0", "tag1"]
+  tags    = ["tag0"]
+}
+
+resource "consul_node" "compute" {
+  name    = "compute-google"
+  address = "www.google.com"
+}
+```
+
+Utilizing an existing known node:
+
+```hcl
+resource "consul_service" "google" {
+  name    = "google"
+  node    = "google"
+  port    = 443
 }
 ```
 
 ## Argument Reference
 
 The following arguments are supported:
 
-* `service_id` - (Optional, string) The ID of the service, defaults to the value of `name`
-  if not supplied.
+* `name` - (Required, string) The name of the service.
+
+* `node` - (Required, string) The name of the node the to register the service on.
 
 * `address` - (Optional, string) The address of the service. Defaults to the
-  address of the agent.
+  address of the node.
 
-* `name` - (Required, string) The name of the service.
+* `service_id` (Optional, string) - If the service ID is not provided, it will be defaulted to the value
+of the `name` attribute.
 
 * `port` - (Optional, int) The port of the service.
 
 * `tags` - (Optional, set of strings) A list of values that are opaque to Consul,
   but can be used to distinguish between services or nodes.
 
+* `datacenter` - (Optional) The datacenter to use. This overrides the datacenter in the
+provider setup and the agent's default datacenter.
 
 ## Attributes Reference
 
 The following attributes are exported:
 
-* `service_id` - The id of the service, defaults to the value of `name`.
+* `id` - The ID of the service.
```

Comment:
I'd highly recommend storing IDs that mean something to users in something other than `id`, because we often have to overload those for import, and it gets your users in the habit of treating `id` as opaque while still making the information accessible.

Reply:
You're totally right, this change was not needed actually, we already expose `service_id` for this purpose. Fixed the docs and updated a test to check my assumption there.